portal/views/user.py

```diff
@@ -1547,6 +1548,12 @@ def unique_email():
         result = match.one()
         if user_id != result.id:
             return jsonify(unique=False)
+
+    # Look out for "masked" emails, as they'll create collisions down the road
+    masked = INVITE_PREFIX + email
+    match = User.query.filter(func.lower(User.email) == masked.lower())
+    if match.count():
```

Comment:
Might make sense to move to the top of the function, seems easier to check and independent of a lot of other conditions.

Reply:
Intentionally inserted after the above logic, handling complexity of the user changing his/her own password.